src/react-vite/12-testing/testing.md

```diff
@@ -38,71 +38,77 @@ React Testing Library’s north star is:
-
-The library promotes accessibility in React applications by providing tools and encouraging patterns
-that make it easier to create accessible web interfaces. By using semantic HTML and ARIA roles in
-tests, developers are nudged towards building applications that are more accessible to all users,
-including those using assistive technologies.
-
-> 3. Utility implementations and APIs should be simple and flexible.
-
-React Testing Library is designed to be lightweight and straightforward, avoiding the complexity that
-can sometimes come with extensive testing frameworks. It encourages good testing practices, such as
-avoiding testing the internal state of components and instead focusing on what the user would see
-and do. The library’s API is intentionally limited to encourage tests that align with these principles.
-
-### Rendering and verifying a component in a test
+rather than the implementation details of the components.
 
-Let’s take a look at some code that we added in [Handling User Inputs and
-Forms](./controlled-vs-uncontrolled.html).
+### <span id="rendering-and-verifying">Rendering and verifying a component in a test</span>
 
-@sourceref ../../../exercises/react-vite/11-controlled-vs-uncontrolled/03-problem/src/components/FormTextField/FormTextField.test.tsx
-@highlight 10-15, only
+OK, you're convinced that testing-library is a good idea, how do you use it to create tests for your
+React components?
 
-Each test consists of arguments to the Vite provided test function, `it`. The first argument is a
-short description of what the test is examining. The convention is that the description string takes
-"it" as a prefix and proceeds from there, e.g. "(it) renders with correct label and type."
+Let's take a look at an example; say we want to test a component we created named `FormTextField`
+that renders a form field. We expect that the component will generate HTML that looks like this:
 
-The second argument is a callback function that runs the test. In the example above the first line
-of the callback invokes `render` and passes it a single argument, JSX for the `FormTextField`
-component including props. After `render` completes the `screen` (a test DOM) will contain the
-elements created by our React code.
+```html
+<div>
+  <label htmlFor="inputId">Phone:</label>
+  <input id="inputId" type="tel" value="555-555-5555" />
+</div>
+```
 
-@sourceref ../../../exercises/react-vite/11-controlled-vs-uncontrolled/03-problem/src/components/FormTextField/FormTextField.test.tsx
-@highlight 11, only
+We want to test `FormTextField` to make sure it generates the DOM we expect. If you're already
+familiar with testing frontend JavaScript code the following pattern will probably be recognizable:
+each test consists of arguments to the Vite provided test function, `it`. The first argument is a
```

Comment:
I think this would read better if it were changed to:

> each test consists of arguments to the `it` function provided by Vite.

Reply:
fixed in next PR